app/src/main/java/com/nsoft/github/domain/usecase/NetworkingUseCase.kt

```diff
@@ -126,6 +126,54 @@ abstract class NetworkingUseCase<
                                     callParams
                                 ).execute()
                             }
+
+                            ApiCallType.PATH -> {
+                                // Ughhhhh this hack .....
+                                // The map doesn't preserve ordering. So, hopefully we get lucky by
+                                // taking the e.g. query map's values and hope they turn out alright
+                                //TODO sort them lexicographically then invert if this doesn't work out though ...
+                                val pathParams = convertedParams.third.values
+                                val callParams = CallParams.PathParams(
+                                    pathParams = pathParams.toList()
+                                )
```

Comment:
like mentioned before, "order" these by assigning them to predefined MAP keys, then pull them out from the map and put them in the list, in-order. then there's none of this "hack" anymore, and the only 'dirty' thing is using the QueryMap as the mule...

Reply:
also, extract this whole 'extraction logic' into a private method.

```
val pathParamsList = extractPathParamsListFromMap(convertedParams.third)
val callParams = CallPArams.PathParams(
    pathParams = pathParamsList
)
```

Reply:
yep, done

Reply:
resolving.